src/cargo/sources/git/utils.rs

```diff
@@ -1067,11 +1136,8 @@ fn github_fast_path(
     config: &Config,
 ) -> CargoResult<FastPathRev> {
     let url = Url::parse(url)?;
-    if !is_github(&url) {
-        return Ok(FastPathRev::Indeterminate);
-    }
```

Comment:
How come this was moved further down?

Reply:
Because the early return  [here](https://github.com/rust-lang/cargo/pull/11176/files/2c4e9f09cc105921e9f112fcf0d021c311fe9277..0270a4e2109db8e22fcd75cbaab899d31f82720f#diff-0e22d3c91f7eac28d92382b445103b421ef842eef760bdc1885be073117f3e9dR1169) does not need github.

Reply:
For some reason that link isn't working.  I'm assuming you are referring to the `is_short_hash_of` check.  I had assumed that is mostly relying on the rest of the github fast path.  In looking again, I'm guessing it isn't?  If so, we should probably clarify these fast paths as this is listed in "github_fast_path" which implies the is_github check should be first.

as a "nit", I would have preferred this to be in a separate commit.  I was looking at this, assuming this was related to the rest of the commit which helped lead to my confusion.  That'd also help with future questions via `git blame` and when debugging with `git bisect`

Reply:
Ooops the link does not work, sorry. Yes I was referring to the `is_short_hash_of`. I will remove in the next rev. If we decide to reorganize the fast paths, it should be its own PR, not just commit.